### PR TITLE
Update Sidecar requirements documentation.

### PR DESCRIPTION
**Pull request by malcyon**
I have not released Graylog Sidecar `1.1.0` yet, but when I do, I'd like the documentation to be clear on the new server requirements (See https://github.com/Graylog2/collector-sidecar/pull/391). 

This should not be merged yet (not until Sidecar `1.1.0` is released). But I would like feedback on how clear this looks. Does it make sense? 

I think the idea is that if Sidecar installs Beats 7.0, that older versions of Graylog won't work with it.

### CODE REVIEW

File: pages/sidecar.rst [hunk 1]
```diff
@@ -28,15 +28,17 @@ You can get .deb and .rpm packages for Graylog Sidecar in our package repository
...
-| 1.0.x           | 3.0.x,3.1.x,3.2.x                |
-+-----------------+----------------------------------+
-| 0.1.x           | 2.2.x,2.3.x,2.4.x,2.5.x,3.0.x    |
-+-----------------+----------------------------------+
-| 0.0.9           | 2.1.x                            |
-+-----------------+----------------------------------+
++-------------------+-----------------------------------+
+| Sidecar version   | Graylog server version            |
++===================+===================================+
+| 1.1.x             | 3.2.5 or higher, 4.0 or higher    |
```
Comment (mpfz0r) [inline pages/sidecar.rst]: Graylog 3.3.x also supports the new beats collectors.
I'm not sure how explicit we should be here.
I'd say `3.2.5 or higher` covers everything

**Comment by mpfz0r** [comment]
> I think the idea is that if Sidecar installs Beats 7.0, that older versions of Graylog won't work with it.

Just to clarify: 
This isn't really a sidecar incompatibility, but a beats one.
It only affects the Windows sidecar, because that's the only one where we are shipping the beats binaries.
On Linux, users are responsible themselves to install beats. And they could've already updated beats to 7.x  and experienced problems.

And I'm not suggesting we should document any of that ;-)

File: pages/sidecar.rst (hunk 1 above)
Comment (malcyon) [reply pages/sidecar.rst]: Good point!

**Comment by malcyon** [comment]
@mpfz0r Thanks for the clarification!

**Review by mpfz0r — APPROVED** [review]
Looks good. I guess we should also apply that change to the older doc branches. like at least 3.3 and 3.2.

**Comment by malcyon** [comment]
> Looks good. I guess we should also apply that change to the older doc branches. like at least 3.3 and 3.2.

Right. I'll take care of that, too.

**Review by dulanism — APPROVED** [review]
+1. Thank you, @juju2112

**Pull request merged**